src/main/java/com/factset/sdk/utils/authentication/ConfidentialClient.java

```diff
@@ -156,6 +186,39 @@ public String getAccessToken() throws AccessTokenException, SigningJwsException
         return this.fetchAccessToken();
     }
 
+    private void requestProviderMetadata() throws AuthServerMetadataContentException, AuthServerMetadataException {
+        LOGGER.debug("Attempting to get response from Well Known URI");
+        URL wellKnownURL = this.config.getWellKnownUrl();
+        InputStream stream;
+
+        try {
+            if (this.requestOptions == null) stream = wellKnownURL.openStream();
+            else {
+                HttpURLConnection conn = (HttpURLConnection) wellKnownURL.openConnection(this.requestOptions.getProxy());
+                if (conn instanceof HttpsURLConnection) {
+                    HttpsURLConnection sslConn = (HttpsURLConnection) conn;
+                    sslConn.setHostnameVerifier(this.requestOptions.getHostnameVerifier());
+                    sslConn.setSSLSocketFactory(this.requestOptions.getSslSocketFactory());
+                }
+
+                stream = conn.getInputStream();
```

Comment:
... .getWellKnownUrl().openConnection( ... proxy stuff ...).getContent().toString();

this could word as well (didn't tested it) but the we dont need a input stream

Reply:
Will look into this as I actually test the request on a proxy server.

Reply:
i forgot the fyi.

Reply:
I will leave this for now as it is not essential and would require making changes to the tests as well.